client/src/components/TripElement/TripEntry.js

```diff
@@ -86,61 +107,131 @@ export function TripEntry({ id, buttonClassName, trip }) {
-            value={tripName}
-            onChange={handleInputChange}
-          />
-        )}
+        <div className='flex w-[160px] justify-start space-x-2'>
+          <PinDropOutlinedIcon className='align-baseline' />
+          {!isRenaming ? (
+            <div className='w-[160px] truncate text-start'>{trip.tripName}</div>
+          ) : (
+            <input
+              ref={inputRef}
+              className='w-[150px] items-center bg-green-200/40 font-medium'
+              placeholder={trip.tripName}
+              value={tripName}
+              onChange={handleInputChange}
+            />
+          )}
+        </div>
       </Button>
-      <div className='absolute right-0 top-0 flex flex-row'>
+    ),
+    [
+      dispatch,
+      id,
+      trip,
+      buttonClassName,
+      isRenaming,
+      tripName,
+      handleInputChange,
+    ],
+  )
+
+  const extraButtons = useMemo(
+    () => (
+      <div
+        id={`extra-buttons-for-${id}`}
+        className={`absolute right-0 top-[9px] flex w-[68px] flex-row text-white`}
+      >
         {!isRenaming ? (
           <>
             <Button
               key={`sidebar-trip-button-${id}-rename`}
               onClick={() => {
                 setIsRenaming(true)
               }}
-              className={`${isSelected} px-0 hover:text-red-400`}
+              className={`${isSelected} h-[22px] w-[22px] hover:text-red-400`}
+              padding='p-0'
             >
-              <EditOutlinedIcon className='align-baseline' />
+              <EditOutlinedIcon
+                sx={{ fontSize: 22 }}
+                className='align-baseline'
+              />
             </Button>
             <Button
               key={`sidebar-trip-button-${id}-edit`}
               onClick={() => {
                 dispatch(closeSidebar())
                 dispatch(openEditTripModal())
               }}
-              className={`${isSelected} hover:text-red-400`}
+              className={`${isSelected} h-[22px] w-[22px] hover:text-red-400`}
+              padding='p-0'
+            >
+              <SettingsOutlinedIcon
+                sx={{ fontSize: 22 }}
+                className='align-baseline'
+              />
+            </Button>
+            <Button
+              key={`sidebar-trip-button-${id}-delete`}
+              onClick={proceedToDelete}
+              className={`${isSelected} h-[22px] w-[22px] hover:text-red-400`}
+              padding='p-0'
             >
-              <SettingsOutlinedIcon className='align-baseline' />
+              <DeleteForeverOutlinedIcon
+                sx={{ fontSize: 22 }}
+                className='align-baseline'
+              />
             </Button>
+
+            <ConfirmationDialog
+              open={openDialog}
+              handleClose={closeDialog}
+              handleConfirm={handleDeleteTrip}
+              title='Confirm Deletion'
+              description='Are you sure you want to delete this trip?'
```

Comment:
is it possible to include the title of the trip they're deleting here?

Reply:
yeah, i think including it in the title works. i can look into it